extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/AbstractDataSourceProducer.java

```diff
@@ -195,6 +196,21 @@ public boolean isValid(Connection connection) {
             poolConfiguration.maxLifetime(dataSourceRuntimeConfig.maxLifetime.get());
         }
 
+        // CloudSQL config
+        if (dataSourceRuntimeConfig.useGcp) {
+            disableSslSupport();
+            String cloudSqlInstance = dataSourceRuntimeConfig.cloudSqlInstance.orElseThrow(
+                    () -> new RuntimeException("Cloud sql instance property is mandatory to use Gcp Cloudsql"));
+            if (cloudSqlInstance.split(":").length != 3) {
+                throw new RuntimeException("Cloud sql instance should match the pattern project-id:zone:cloudSqlInstance");
+            }
+            agroalConnectionFactoryConfigurationSupplier.jdbcUrl(
+                    MessageFormat.format("{0}?socketFactory=com.google.cloud.sql.postgres.SocketFactory&cloudSqlInstance={1}"
```

Comment:
The socketFactory is bound to postgres driver and if you would like to use a different driver is not possible with current implementation. Is it possible to make this configurable from a property as well ?

Reply:
Currently there are only two supported SGBD postgres and mysql, but yes I should make this configurable.

Reply:
But I think @Sanne is correct, this implementation is very GCP integrated. maybe we should switch a little bit the way we make the implementation for this.

I think the easiest way is to use the `agroalConnectionFactoryConfigurationSupplier.jdbcProperty(<property_name>, <value>)` method

Reply:
I tried this but it didn't worked. I will re-take a loot at it. 

In fact I think I should create a separate extension for gcp sql datasources.

Reply:
Can you check: https://github.com/quarkusio/quarkus/pull/6634 ?

Reply:
I think it won't work (i tried this already)